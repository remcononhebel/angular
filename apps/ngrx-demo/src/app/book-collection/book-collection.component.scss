.book-collection {
  &__item {
    margin-top: 20px;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
    padding: 16px;
    display: flex;
    flex-direction: column;

    &-add {
      display: flex;
      justify-content: end;
    }
  }
}
