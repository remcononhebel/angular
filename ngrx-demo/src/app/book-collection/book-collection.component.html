<div class="book-collection">
  <div class="book-collection__item" *ngFor="let book of books">
    <h2 class="book-collection__item-title">
      {{ book.volumeInfo.title }}
    </h2>
    <div class="book-collection__item-authors">by {{ book.volumeInfo.authors }}</div>
    <div class="book-collection__item-add">
      <button (click)="remove.emit(book.id)" data-test="remove-button">
        Remove from Collection
      </button>
    </div>
  </div>
</div>
