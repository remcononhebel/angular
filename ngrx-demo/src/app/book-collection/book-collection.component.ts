import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Book } from '../model/book.model';

@Component({
  selector: 'app-book-collection',
  templateUrl: './book-collection.component.html',
  styleUrls: ['./book-collection.component.scss'],
})
export class BookCollectionComponent {
  @Input() books: Book[] = [];
  @Output() remove = new EventEmitter<string>();
}
