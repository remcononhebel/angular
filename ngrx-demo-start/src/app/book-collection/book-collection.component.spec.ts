import { ComponentFixture } from '@angular/core/testing';
import { Spectator, createComponentFactory } from '@ngneat/spectator';
import { BookCollectionComponent } from './book-collection.component';

describe('BookCollectionComponent', () => {
  let spectator: Spectator<BookCollectionComponent>;
  let component: BookCollectionComponent;
  let fixture: ComponentFixture<BookCollectionComponent>;
  const createComponent = createComponentFactory(BookCollectionComponent);

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
    fixture = spectator.fixture;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
