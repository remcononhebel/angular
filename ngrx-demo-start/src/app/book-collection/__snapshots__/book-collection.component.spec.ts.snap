// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCollectionComponent should match snapshot 1`] = `
<app-book-collection
  books={[Function Array]}
  remove={[Function EventEmitter_]}
>
  <div
    class="book-collection"
  >
  </div>
</app-book-collection>
`;
